src/Tools/onnx-mlir-opt/onnx-mlir-opt.cpp

```diff
@@ -104,6 +109,10 @@ int main(int argc, char **argv) {
 
   registry.insert<mlir::ONNXDialect>();
   registry.insert<mlir::KrnlOpsDialect>();
+#ifdef __NNPA__
```

Comment:
Could we place these dialect insertions into the registry inside  a function (suggest `registerNNPAPasses()`). Ideally we should avoid the use preprocessor macros all over the place and have something like:

```
   if (NNPAAccellerator.isActive()) 
      registerNNPAPasses();
 ```
 
 I am concerned that over time, as more accelerators are added, we would end up with a bunch of macro checks, making the code less readable.

Reply:
yes, see my comment above.